src/hooks/usePlayList.js

```diff
@@ -1,87 +1,64 @@
+// @ts-check
 import { useEffect, useState } from 'react'
 import getPlaylistData from '../utils/getPlaylistData'
+import getWrongAnswers from '../utils/getWrongAnswers'
 import shuffle from '../utils/shuffleArray'
 
 export default function usePlayList(playlist) {
-  const newPlaylist = shuffle(playlist)
+  const newPlaylist = playlist
   const [playlistData, setPlaylistData] = useState(null)
   const [answers, setAnswers] = useState(null)
   const [counter, setCounter] = useState(null)
   const [wrongAnswers, setWrongAnswers] = useState(null)
   let getNextUrl
 
-  // useEffect(()=> {
-
-  // })
-
-  //get playlist data
+  // get playlist data
   useEffect(() => {
-    const baseUrl = '/api/playlist/'
-    Promise.all(
-      newPlaylist.map(({ id }) => fetch(baseUrl + id).then(res => res.json()))
-    ).then(data => {
-      const newPlaylistData = data.map(({ results }) => results[0])
-      const cleanPlaylistData = newPlaylistData.filter(el => !!el)
-      !playlistData && setPlaylistData(cleanPlaylistData)
-      setCounter(cleanPlaylistData.length - 1)
-    })
-  }, [])
+    async function getData() {
+      const data = await getPlaylistData(shuffle(newPlaylist))
+      if (data) {
+        setPlaylistData(data)
+        setCounter(data?.length - 1)
+      }
+    }
+    getData()
```

Comment:
You ran out of function names, right? A different solution could be an _IIFE_.

Reply:
To be honest, that's exactly what I tried. But for some reason it didn't work, so I thought it was the useEffect. But I will try it again then!